Dockerfile

```diff
@@ -5,11 +5,13 @@ ARG DATABASE_TYPE
 ENV DATABASE_URL "postgresql://umami:umami@db:5432/umami" \
     DATABASE_TYPE=$DATABASE_TYPE
 
-COPY . /app
 WORKDIR /app
+EXPOSE 3000
 
-RUN npm install && npm run build
+COPY package.json yarn.lock /app/
+RUN yarn install --frozen-lockfile
 
-EXPOSE 3000
+COPY . /app
```

Comment:
This is a nit, but we could probably just do one `COPY` here and then a single `RUN` instead of having them on separate layers:

```dockerfile
COPY . /app
RUN yarn install --frozen-lockfile \
      && yarn build
```

Reply:
we can, but usually two layers is a better option.

cause' when dependency changed rebuild is required,
but if only logic level changes, dependecy layer caching can speed the building a lot.

ref: https://buddy.works/guides/how-speed-up-docker-build

Reply:
I agree in the case where you're using a Dockerfile for development, but for production images you're likely not caching the dependencies anyway (e.g. in CI), so doesn't make sense to split them across layers.

That said, the increase in image size here from the extra layer is minuscule, so it's really fine either way! 👍🏻